File: codeup/100000575/A.cpp
#include <iostream>
using namespace std;

int main() {
  int L, M, tree_list[10002], a, b, sum;
  while (scanf("%d %d", &L, &M) != EOF) {
    if (L == 0 && M == 0) break;
    fill(tree_list, tree_list + L + 1, 1);
    while (M--) {
      scanf("%d %d", &a, &b);
      for (int i = a; i <= b; i++) tree_list[i] = 0;
    }
    sum = 0;
    for (int i = 0; i <= L; i++) sum += tree_list[i];
    printf("%d\n", sum);
  }
  return 0;
}

File: codeup/100000575/A.py
#!/usr/bin/python3
while True:
    L, M = [int(x) for x in input().split()]
    if L == 0 and M == 0:
        break
    tree_list = [1 for i in range(L+1)]
    for i in range(M):
        a, b = [int(x) for x in input().split()]
        for iter in range(a, b+1):
            tree_list[iter] = 0
    sum = 0
    for num in tree_list:
        sum += num
    print(sum)
